skimage/filters/_frangi.py

```diff
@@ -0,0 +1,185 @@
+import numpy as np
+
+
+__all__ = ['frangi_filter', 'hessian_filter']
+
+
+def frangi_filter(image, scale=(1, 10), scale_ratio=2, beta1=0.5, beta2=15,
+                  black_white=True):
+    """Returns image filtered with frangi filter.
+
+    Calculates the eigenvectors of the Hessian to compute the likeliness of
+    an image region to vessels, according to the method described in _[1].
+
+    Parameters
+    ----------
+    image : (N, M) ndarray
+        Array with input image data.
+    scale : tuple, optional
+        The range of sigmas used, default [1 10].
+    scale_ratio : int, optional
+        Step size between sigmas, default 2.
+    beta1 : float, optional
+        Frangi correction constant, default 0.5.
+    beta2 : float, optional
+        Frangi correction constant, default 15.
+    black_white : boolean, optional
+        Detect black ridges (default) set to true, for
+        white ridges set to false.
+
+    Returns
+    -------
+    J : (N, M) ndarray
+        Filtered image (maximum of pixels across all scales).
+
+    Notes
+    -----
+    Written by Marc Schrijver, 2/11/2001
+    Re-Written by D.Kroon University of Twente (May 2009)
+
+    References
+    ----------
+    .. [1] A. Frangi, W. Niessen, K. Vincken, and M. Viergever. "Multiscale
+           vessel enhancement filtering," In LNCS, vol. 1496, pages 130-137,
+           Germany, 1998. Springer-Verlag.
+    .. [2] Kroon, D.J.: Hessian based frangi vesselness filter.
+    .. [3] http://mplab.ucsd.edu/tutorials/gabor.pdf.
+    """
+
+    from ..feature import hessian_matrix, hessian_matrix_eigvals
+    sigmas = range(scale[0], scale[1], scale_ratio)
```

Comment:
For the first question, if yes, then change the value of input parameters to include float, and use np.arange here.

For the second one, more intuitive names would be `scale_range` and `scale_step`, or `scale_start`, `scale_end`, `scale_step`. (Potentially, rename scale to sigma.)